python/sbp/client/loggers/rotating_logger.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (C) 2015 Swift Navigation Inc.
+# Contact: Bhaskar Mookerji <mookerji@swiftnav.com>
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+from logging.handlers import TimedRotatingFileHandler
+from sbp.client.loggers.json_logger import JSONLogger
+from sbp.table import dispatch
+import json
+import logging
+import time
+import warnings
+
+# TODO (Buro): Do we need to have a queue or some special threading
+# here? Test to see that the LRU messages during a time interval
+# actually made it into the log, didn't get lost, or cause other
+# blocking issues.
+
+
+class RotatingFileLogger(JSONLogger):
+  """RotatingFileLogger
+
+  The :class:`RotatingFileLogger` JSON logs SBP messages to a rotating
+  file handler that's turned over at a specified time interval
+  (defaults to 30 minutes). Intended to be a very low-overhead,
+  cross-platform rolling logger for field testing.
+
+  Parameters
+  ----------
+  filename : string
+    Path to file to write SBP messages to.
+  when : str
+    Specifies a type of interval
+  interval : int
+    Specifies length of interval
+  backupCount : int
+    Number of backups to keep around
+  mode : str
+    File mode
+  tags : dict
+    Tags to add to SBP message envelope
+  dispatcher: dict
+    SBP dispatch table
+
+  """
+
+  def __init__(self, filename, when='M', interval=30, backupCount=3,
+               mode="w", tags={}, dispatcher=dispatch):
+    self.handler = TimedRotatingFileHandler(filename, when, interval, backupCount)
+    self.logger = logging.getLogger("Rotating Log")
+    self.logger.setLevel(logging.INFO)
+    self.logger.addHandler(self.handler)
+    self.dispatcher = dispatcher
+    self.base_time = time.time()
+    self.tags = tags
+
+  def __call__(self, msg):
+    self.call(msg)
+
+  def flush(self):
+    self.handler.flush()
+
+  def close(self):
+    self.handler.close()
+
+  def call(self, msg):
+    try:
+      self.logger.info(json.dumps(self.fmt_msg(msg), allow_nan=False))
+    except ValueError:
```

Comment:
Man, maybe we should move this error handling logic into a dumps message on the base class? I think we're starting to repeat it in a couple of places. There's got to be a better way to just change the the implementation of how the string goes out here.